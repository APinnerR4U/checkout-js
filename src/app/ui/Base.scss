@import '../../../node_modules/@bigcommerce/citadel/src/tools/toolkit';
@import '../../scss/tools/toolkit';
@import '../../scss/settings/global/global';
@import '../../../node_modules/@bigcommerce/citadel/src/settings/global/global';
@import '../../../node_modules/@bigcommerce/citadel/src/settings/normalize/normalize';
@import '../../../node_modules/@bigcommerce/citadel/src/settings/foundation/foundation';
@import '../../../node_modules/@bigcommerce/citadel/src/settings/bigcommerce/bigcommerce';
@import '../../scss/settings/foundation/foundation';
@import '../../scss/settings/normalize/normalize';
@import '../../scss/settings/utilities/utilities';
@import '../../scss/settings/bigcommerce/bigcommerce';
@import '../../scss/settings/checkout/checkout';
@import '../../../node_modules/@bigcommerce/citadel/src/utilities/display/tools';
@import '../../../node_modules/@bigcommerce/citadel/src/vendor/foundation/foundation.scss';

$global-radius: 4px;
