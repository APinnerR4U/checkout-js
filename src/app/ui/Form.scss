@import "./Base.scss";

.form {
    @include breakpoint("large") {
        max-width: remCalc(540px);
    }
}

.checkout-form {
    margin: 0;

    @include breakpoint("small") {
        margin-left: remCalc($checkoutStep-counter-size) + $checkoutStep-header-figure-margin;
    }

    @include breakpoint("large") {
        margin-left: 0;
    }

    @include breakpoint(1080px) {
        margin-left: remCalc($checkoutStep-counter-size) + $checkoutStep-header-figure-margin;
    }
}

.form-legend-container {
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.form-actions {
    margin: #{spacing("single") + spacing("half")} 0;
    padding: 0;

    .button:first-child {
        margin-left: 0;
    }
}

@media print {
    .guest-signup {
        display: none;
    }
}
