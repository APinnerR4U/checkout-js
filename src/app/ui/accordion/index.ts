import { AccordionProps } from './Accordion';
import { AccordionItemHeaderProps, AccordionItemProps } from './AccordionItem';

export type AccordionProps = AccordionProps;
export type AccordionItemHeaderProps = AccordionItemHeaderProps;
export type AccordionItemProps = AccordionItemProps;

export { default as Accordion } from './Accordion';
export { default as AccordionItem } from './AccordionItem';
