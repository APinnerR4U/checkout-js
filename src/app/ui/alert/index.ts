import { AlertProps } from './Alert';
import { FlashAlertProps, FlashMessage } from './FlashAlert';

export type AlertProps = AlertProps;
export type FlashAlertProps = FlashAlertProps;
export type FlashMessage = FlashMessage;

export { default as Alert, AlertType } from './Alert';
export { default as FlashAlert } from './FlashAlert';
