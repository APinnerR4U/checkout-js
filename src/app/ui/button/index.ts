import { ButtonProps } from './Button';

export type ButtonProps = ButtonProps;

export { default as Button, ButtonSize, ButtonVariant } from './Button';
