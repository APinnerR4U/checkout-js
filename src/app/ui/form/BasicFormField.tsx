import { getIn, Field, FieldConfig, FieldProps } from 'formik';
import { isDate, noop } from 'lodash';
import React, { createElement, Component, FunctionComponent } from 'react';

import FormFieldContainer from './FormFieldContainer';

export interface BasicFormFieldProps extends FieldConfig {
    additionalClassName?: string;
    className?: string;
    testId?: string;
    onChange?(value: any): void;
}

const BasicFormField: FunctionComponent<BasicFormFieldProps> = ({
    additionalClassName,
    className,
    component,
    render,
    testId,
    onChange,
    ...rest
}) => <Field
    { ...rest }
    render={ (props: FieldProps) => {
        const {
            field,
            form: { errors },
        } = props;

        return (
            <FormFieldContainer
                additionalClassName={ additionalClassName }
                testId={ testId }
                hasError={ getIn(errors, field.name) }
            >
                <FieldInput
                    { ...props }
                    onChange={ onChange }
                    component={ component }
                    render={ render }
                />
            </FormFieldContainer>
        );
    } }
/>;

type FieldInputProps = FieldProps & Pick<FieldConfig, 'component' | 'render'> & {
    onChange?(value: string): void;
};

class FieldInput extends Component<FieldInputProps> {
    componentDidUpdate({ field: prevField }: FieldInputProps) {
        const { field: { value }, onChange = noop } = this.props;
        const comparableValue = isDate(value) ? value.getTime() : value;
        const comparablePrevValue = isDate(prevField.value) ? prevField.value.getTime() : prevField.value;

        if (comparableValue !== comparablePrevValue) {
            onChange(value);
        }
    }

    render() {
        const {
            component = 'input',
            field,
            render,
        } = this.props;

        if (render) {
            // tslint:disable-next-line:no-unnecessary-type-assertion
            return (render as any)(this.props);
        }

        if (typeof component === 'string') {
            // tslint:disable-next-line:no-unnecessary-type-assertion
            return createElement(component as any, field);
        }

        // tslint:disable-next-line:no-unnecessary-type-assertion
        return createElement(component as any, this.props);
    }
}

export default BasicFormField;
