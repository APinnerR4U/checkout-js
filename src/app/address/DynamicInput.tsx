import { FormFieldItem } from '@bigcommerce/checkout-sdk';
import { isDate, noop } from 'lodash';
import React, { memo, useCallback, FunctionComponent } from 'react';
import ReactDatePicker from 'react-datepicker';

import { CheckboxInput, InputProps, RadioInput, TextArea, TextInput } from '../ui/form';

import DynamicFormFieldType from './DynamicFormFieldType';

export interface DynamicInputProps extends InputProps {
    id: string;
    additionalClassName?: string;
    value?: string | string[];
    rows?: number;
    fieldType?: DynamicFormFieldType;
    options?: FormFieldItem[];
}

const DynamicInput: FunctionComponent<DynamicInputProps> = ({
    additionalClassName,
    fieldType,
    id,
    name,
    onChange = noop,
    options,
    placeholder,
    value,
    ...rest
}) => {
    const handleDateChange = useCallback((date, event) => onChange({
        ...event,
        target: {
            name,
            value: date,
        },
    }), [
        onChange,
        name,
    ]);

    switch (fieldType) {
    case DynamicFormFieldType.dropdown:
        return (
            <select
                { ...rest as any }
                name={ name }
                onChange={ onChange }
                id={ id }
                data-test={ `${id}-select` }
                className="form-select optimizedCheckout-form-select"
                value={ value === null ? '' : value }
            >
                { placeholder &&
                    <option value="">
                        { placeholder }
                    </option> }
                { options && options.map(({ label, value: optionValue }) =>
                    <option
                        key={ optionValue }
                        value={ optionValue }
                    >
                        { label }
                    </option>
                ) }
            </select>
        );

    case DynamicFormFieldType.radio:
        if (!options || !options.length) {
            return null;
        }

        return <>{ options.map(({ label, value: optionValue }) =>
            <RadioInput
                { ...rest }
                name={ name }
                onChange={ onChange }
                id={ `${id}-${optionValue}` }
                testId={ `${id}-${optionValue}-radio` }
                key={ optionValue }
                label={ label }
                value={ optionValue }
                checked={ optionValue === value }
            />) }</>;

    case DynamicFormFieldType.checkbox:
        if (!options || !options.length) {
            return null;
        }

        return <>{ options.map(({ label, value: optionValue }) =>
            <CheckboxInput
                { ...rest }
                name={ name }
                onChange={ onChange }
                id={ `${id}-${optionValue}` }
                testId={ `${id}-${optionValue}-checkbox` }
                key={ optionValue }
                label={ label }
                value={ optionValue }
                checked={ Array.isArray(value) ? value.includes(optionValue) : false }
            />) }</>;

    case DynamicFormFieldType.date:
        return (
            <ReactDatePicker
                { ...rest as any }
                name={ name }
                // FIXME: we can avoid this by simply using onChangeRaw, but it's not being triggered properly
                // https://github.com/Hacker0x01/react-datepicker/issues/1357
                // onChangeRaw={ rest.onChange }
                onChange={ handleDateChange }
                autoComplete="off"
                placeholderText="MM/DD/YYYY"
                minDate={ rest.min ? new Date(rest.min) : undefined }
                maxDate={ rest.max ? new Date(rest.max) : undefined }
                className="form-input optimizedCheckout-form-input"
                popperClassName="optimizedCheckout-contentPrimary"
                calendarClassName="optimizedCheckout-contentPrimary"
                selected={ isDate(value) ? value : undefined }
            />
        );

    case DynamicFormFieldType.multiline:
        return (
            <TextArea
                { ...rest as any }
                name={ name }
                onChange={ onChange }
                id={ id }
                testId={ `${id}-text` }
                type={ fieldType }
                value={ value }
            />
        );

    default:
        return (
            <TextInput
                { ...rest }
                name={ name }
                onChange={ onChange }
                id={ id }
                testId={ `${id}-${ fieldType === DynamicFormFieldType.password ?
                    'password' :
                    'text' }` }
                type={ fieldType }
                value={ value }
            />
        );
    }
};

export default memo(DynamicInput);
