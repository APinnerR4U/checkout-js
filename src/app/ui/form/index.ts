import { BasicFormFieldProps } from './BasicFormField';
import { CheckboxFormFieldProps } from './CheckboxFormField';
import { ChecklistProps } from './Checklist';
import { ChecklistItemProps } from './ChecklistItem';
import { ChecklistItemInputProps } from './ChecklistItemInput';
import { FieldsetProps } from './Fieldset';
import { FormProps } from './Form';
import { FormFieldProps } from './FormField';
import { FormFieldErrorProps } from './FormFieldError';
import { FormContextType } from './FormProvider';
import { InputProps } from './Input';
import { LabelProps } from './Label';
import { LegendProps } from './Legend';
import { TextAreaProps } from './TextArea';
import { TextInputProps } from './TextInput';

export type FieldsetProps = FieldsetProps;
export type FormProps = FormProps;
export type FormFieldProps = FormFieldProps;
export type BasicFormFieldProps = BasicFormFieldProps;
export type CheckboxFormFieldProps = CheckboxFormFieldProps;
export type ChecklistProps = ChecklistProps;
export type ChecklistItemProps = ChecklistItemProps;
export type FormContextType = FormContextType;
export type FormFieldErrorProps = FormFieldErrorProps;
export type InputProps = InputProps;
export type TextAreaProps = TextAreaProps;
export type TextInputProps = TextInputProps;
export type LabelProps = LabelProps;
export type LegendProps = LegendProps;
export type ChecklistItemInputProps = ChecklistItemInputProps;

export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormProvider, FormContext } from './FormProvider';
export { default as FormField } from './FormField';
export { default as BasicFormField } from './BasicFormField';
export { default as CheckboxFormField } from './CheckboxFormField';
export { default as Checklist } from './Checklist';
export { default as ChecklistItem } from './ChecklistItem';
export { default as FormFieldError } from './FormFieldError';
export { default as FormFieldContainer } from './FormFieldContainer';
export { default as Input } from './Input';
export { default as TextArea } from './TextArea';
export { default as TextInput } from './TextInput';
export { default as TextInputIframeContainer } from './TextInputIframeContainer';
export { default as RadioInput } from './RadioInput';
export { default as CheckboxInput } from './CheckboxInput';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as ChecklistItemInput } from './ChecklistItemInput';
