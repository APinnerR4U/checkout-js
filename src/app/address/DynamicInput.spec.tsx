import { shallow } from 'enzyme';
import React from 'react';
import ReactDatePicker from 'react-datepicker';

import { CheckboxInput, RadioInput, TextArea, TextInput } from '../ui/form';

import DynamicFormFieldType from './DynamicFormFieldType';
import DynamicInput from './DynamicInput';

describe('DynamicInput', () => {
    it('renders text input for default input with passed props', () => {
        expect(shallow(<DynamicInput id="field_33" name="x" />).html())
            .toMatchSnapshot();
    });

    it('renders textarea for multiline type', () => {
        expect(shallow(
            <DynamicInput
                id="field_33"
                fieldType={ DynamicFormFieldType.multiline }
                rows={ 4 }
            />)
            .find(TextArea)
            .prop('rows'))
            .toEqual(4);
    });

    it('renders date picker for date type', () => {
        const datePicker = shallow(
            <DynamicInput
                id="field_33"
                fieldType={ DynamicFormFieldType.date }
                min="2019-1-1"
                max="2019-2-1"
            />
        );

        expect(datePicker.find(ReactDatePicker).props())
            .toMatchObject(expect.objectContaining({
                minDate: new Date('2019-01-01 00:00:00'),
                maxDate: new Date('2019-02-01 00:00:00'),
            }));
    });

    it('renders checkbox input for checkbox type', () => {
        const component = shallow(
            <DynamicInput
                fieldType={ DynamicFormFieldType.checkbox }
                id="id"
                value={ ['y'] }
                options={ [
                    { value: 'x', label: 'X' },
                    { value: 'y', label: 'Y' },
                ] }
            />);

        expect(component.find(CheckboxInput).at(0).props())
            .toEqual(expect.objectContaining({
                id: 'id-x',
                checked: false,
            }));

        expect(component.find(CheckboxInput).at(1).props())
            .toEqual(expect.objectContaining({
                id: 'id-y',
                checked: true,
            }));
    });

    it('renders radio type input for radio type', () => {
        const component = shallow(
            <DynamicInput
                fieldType={ DynamicFormFieldType.radio }
                id="id"
                value="y"
                onChange={ jest.fn() }
                options={ [
                    { value: 'x', label: 'X' },
                    { value: 'y', label: 'Y' },
                ] }
            />);

        expect(component.find(RadioInput).at(0).props())
            .toEqual(expect.objectContaining({
                id: 'id-x',
                checked: false,
            }));

        expect(component.find(RadioInput).at(1).props())
            .toEqual(expect.objectContaining({
                id: 'id-y',
                checked: true,
            }));
    });

    it('renders number type input for number type', () => {
        expect(shallow(
            <DynamicInput
                id="field_33"
                fieldType={ DynamicFormFieldType.number }
            />)
            .find(TextInput)
            .prop('type'))
            .toEqual('number');
    });

    it('renders password type input for password type', () => {
        expect(shallow(
            <DynamicInput
                id="field_33"
                fieldType={ DynamicFormFieldType.password }
            />)
            .find(TextInput)
            .prop('type'))
            .toEqual('password');
    });

    it('renders tel type input for phone type', () => {
        expect(shallow(
            <DynamicInput
                id="field_33"
                fieldType={ DynamicFormFieldType.telephone }
            />)
            .find(TextInput)
            .prop('type'))
            .toEqual('tel');
    });

    it('renders select input with passed props', () => {
        expect(shallow(
            <DynamicInput
                id="field_33"
                fieldType={ DynamicFormFieldType.dropdown }
                options={ [
                    { label: 'Foo', value: 'foo'},
                    { label: 'Foo1', value: 'foo1'},
                ] }
                defaultValue="foo"
                placeholder="Select an option"
                name="select"
            />).html())
            .toMatchSnapshot();
    });
});
