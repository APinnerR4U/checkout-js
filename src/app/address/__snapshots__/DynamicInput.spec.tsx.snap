// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicInput renders select input with passed props 1`] = `"<select name=\\"select\\" id=\\"field_33\\" data-test=\\"field_33-select\\" class=\\"form-select optimizedCheckout-form-select\\"><option value=\\"\\">Select an option</option><option selected=\\"\\" value=\\"foo\\">Foo</option><option value=\\"foo1\\">Foo1</option></select>"`;

exports[`DynamicInput renders text input for default input with passed props 1`] = `"<input type=\\"text\\" name=\\"x\\" id=\\"field_33\\" class=\\"form-input optimizedCheckout-form-input\\" data-test=\\"field_33-text\\"/>"`;
